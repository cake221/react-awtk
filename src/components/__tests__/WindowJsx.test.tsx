import * as React from "react";
import {AwtkRender,setParentWidget} from "../.."


describe("button jsx", ()=>{

  beforeEach(()=>{
    __AwtkSnapshot = []
  })

  it("jsx test",()=>{

    AwtkRender(<t_window/>)
    expect(__AwtkSnapshot).toMatchSnapshot()
  })

  it("jsx test2", ()=>{
    AwtkRender(<t_window
        ref = {
          (ref) => setParentWidget(ref, "win1")
        }
      />)
    expect(__AwtkSnapshot).toMatchSnapshot()
  })
})
