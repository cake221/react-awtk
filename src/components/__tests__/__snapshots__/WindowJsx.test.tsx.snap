// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button jsx jsx test 1`] = `
Array [
  "getRootHostContext",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  window_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "finalizeInitialChildren",
  "prepareForCommit",
  "appendNodeChild",
  "resetAfterCommit",
  "getPublicInstance",
]
`;

exports[`button jsx jsx test2 1`] = `
Array [
  "getRootHostContext",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  window_create( parent, x, y, w, h ) :null, 0, 0, 0, 0",
  "finalizeInitialChildren",
  "prepareForCommit",
  "appendNodeChild",
  "resetAfterCommit",
  "getPublicInstance",
  "getPublicInstance",
]
`;
